lib/presentation/bill_details/components/bottom_modal.dart

```diff
@@ -0,0 +1,261 @@
+            _buildPlanCardsRow(),
+            const Gap(16),
+            _buildUpgradeButton(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildPlanCardsRow() {
+    return SingleChildScrollView(
+      scrollDirection: Axis.horizontal,
+      padding: const EdgeInsets.only(left: 16),
+      child: Row(
+        children: [
+          ..._buildPlanCards().map(
+            (card) => Padding(
+              padding: const EdgeInsets.only(right: 8),
+              child: card,
+            ),
+          ),
+          const Gap(16),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildUpgradeButton() {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: WideButton.gradient(
+        label: t.billDetailsPage.upgradeButton,
+        gradient: const LinearGradient(
+          colors: [
+            AppColor.yellow600Primary,
+            AppColor.yellow200,
+          ],
+          begin: Alignment.topLeft,
+          end: Alignment.bottomRight,
+        ),
+        onPressed: () {},
+      ),
+    );
+  }
+
+  List<Widget> _buildPlanCards() {
+    final pricingOptions = t.billDetailsPage.pricingOptions;
+
+    return [
+      _buildPlanCard(
+        PricingOption(
+          duration: pricingOptions.oneDay.duration,
+          price: pricingOptions.oneDay.price,
+          discount: pricingOptions.oneDay.discount,
+        ),
+      ),
+      _buildPlanCard(
+        PricingOption(
+          duration: pricingOptions.threeDays.duration,
+          price: pricingOptions.threeDays.price,
+          discount: pricingOptions.threeDays.discount,
+        ),
+      ),
+      _buildPlanCard(
+        PricingOption(
+          duration: pricingOptions.fiveDays.duration,
+          price: pricingOptions.fiveDays.price,
+          discount: pricingOptions.fiveDays.discount,
+        ),
+      ),
+      _buildPlanCard(
+        PricingOption(
+          duration: pricingOptions.sevenDays.duration,
+          price: pricingOptions.sevenDays.price,
+          discount: pricingOptions.sevenDays.discount,
+        ),
+      ),
+      _buildPlanCard(
+        PricingOption(
+          duration: pricingOptions.lifetime.duration,
+          price: pricingOptions.lifetime.price,
+          discount: pricingOptions.lifetime.discount,
+        ),
+      ),
+    ];
+  }
+
+  Widget _buildPlanCard(PricingOption pricingOption) {
+    return PlanCard(
+      label: pricingOption.duration,
+      price: pricingOption.price,
+      discount: pricingOption.discount,
+      groupValue: _selectedPlan,
+      value: pricingOption.duration,
+      onChanged: (value) {
+        setState(() {
+          _selectedPlan = value!;
+        });
+      },
+    );
+  }
+}
+
+class PricingOption {
+  PricingOption({
+    required this.duration,
+    required this.price,
+    this.discount,
+  });
+  final String duration;
+  final String price;
+  final String? discount;
+}
+
+class PlanCard extends StatelessWidget {
```

Comment:
現状PlanCardというコンポーネントがすでにあるので、こちらを変えてもらえると嬉しい...!
例えば↓(これ以外ももちろん可)
```suggestion
class PurchaseItemCard extends StatelessWidget {
```

Reply:
ワンチャンファイル分けてもいいかも？

Reply:
ファイル分けて名前変えました！Done!